flink-clients/src/main/java/org/apache/flink/client/CliFrontend.java

```diff
@@ -909,7 +919,21 @@ private int handleError(Throwable t) {
 		}
 		LOG.error("Error while running the command.", t);
 
-		t.printStackTrace();
+		// check if the error was due to an invalid program in detached mode.
+		if (t instanceof ProgramInvocationException && t.getCause() instanceof DetachedProgramException) {
+			System.err.println(t.getCause().getMessage());
+			// now trace to the user's main method. We don't wanna show unnecessary information
+			// in this particular case.
```

Comment:
Why in this case? Doesn't that apply to all errors that occur during interactive execution?